srcalc/src/main/java/gov/va/med/srcalc/security/AccessVerifyAuthenticationProvider.java

```diff
@@ -0,0 +1,119 @@
+package gov.va.med.srcalc.security;
+
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+
+import gov.va.med.srcalc.domain.VistaPerson;
+import gov.va.med.srcalc.vista.vistalink.VistaLinkAuthenticator;
+import gov.va.med.srcalc.vista.vistalink.VistaLinkUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.AuthenticationServiceException;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+/**
+ * <p>An AuthenticationProvider implementation that authenticates a user based on a VistA
+ * access/verify code pair.</p>
+ */
+public final class AccessVerifyAuthenticationProvider implements AuthenticationProvider
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            AccessVerifyAuthenticationProvider.class);
+    
+    /**
+     * <p>Authenticates the user with VistA.</p>
+     * 
+     * <p>Note that the returned Authentication is different from the given
+     * Authentication. This is weird, but required by Spring Security.</p>
+     * 
+     * @param providedAuth should be an instance of {@link
+     * UsernamePasswordAuthenticationToken} with the access code as the principal and the
+     * verify code as the credentials.
+     * @returns A fully-populated UsernamePasswordAuthenticationToken if authentication
```

Comment:
There are multiple reasons for why it can return null instead of just having the wrong type.